app/controllers/pledges_controller.rb

```diff
@@ -10,9 +10,11 @@ def new
   end
 
   def create
-    @pledge = @project.pledges.build(pledge_params)
-    if @pledge.save
+    @reward = Reward.find(params[:reward_id])
```

Comment:
something like if request.xhr? for this branch, any then the normal path for people without JS active?